recipes/geotrans/all/conandata.yml

```diff
@@ -0,0 +1,4 @@
+sources:
+  "3.8":
+    url: "https://earth-info.nga.mil/php/download.php?file=wgs-mastertgz"
```

Comment:
The fact they don't have downloads versioned is really problematic. Let's remember:
https://github.com/conan-io/conan-center-index/issues/1779
https://github.com/microsoft/vcpkg/pull/14412
https://github.com/conan-io/conan-center-index/issues/1779

Reply:
I've contacted NGA to request a versioned download link. It's not clear what the release cycle of geotrans actually is, so this may not pose a problem for quite some time. I propose that we move forward with what we have for the moment, and if I get a response from NGA/they do end up providing a versioned link this can be updated in a future PR.

Reply:
This is a blocker, we need reproducibility, we need an URI and the matching hash... otherwise the recipe will start to fail when the sources change.

We can consider storing the sources in [this repository](https://github.com/conan-io/cci-sources-backup), so we have a versioned and inmutable URL.

Reply:
I found this public ftp url: ftp://ftp.nga.mil
Maybe it's worth exploring?

Reply:
I found the files are in `/pub2/gandg/website/wgs84/apps/geotrans/current-version/sw` inside that ftp server.
But - again - only current-versions...

Reply:
I've not received a response from NGA on this, so I've followed it up with a wider audience - hopefully this gets some attention. In the mean time, happy to proceed by mirroring the source in https://github.com/conan-io/cci-sources-backup at your suggestion @jgsogo , if this is the best way to proceed. I'm noting that the repository README indicates that this is a temporary solution?

Another thing to consider here is that the source directory for geotrans contains pre-built binaries and is ~150MB - will this trigger a need for git lfs, and if so is this repo set up to handle that? I'm not sure how well git handles files of this size without lfs

Reply:
Looks like the source size _is_ too large - files over 100MB need to use git LFS. I attempted to push this up to my fork, and i get the error:
> batch response: @samuel-emrys can not upload new objects to public fork samuel-emrys/cci-sources-backup
which appears to be an existing issue relating to no support for git LFS on public forks, see: https://github.com/git-lfs/git-lfs/issues/1906. Regardless, GitHub has some pretty restrictive quotas on LFS usage for their free tier - only up to 1GB of bandwidth, so nothing really worth considering.

I thought that I might be able to circumvent this by repackaging the sources only, but it looks like the vast majority of this is the 143MB occupied by a single data file, which is necessary to use the library.

As I mentioned previously, it's not clear how frequently this library gets updated. Admittedly it's not ideal that the package may break if/when they do release a new version, but surely it's better to have a package that breaks than one that isn't available at all?

Otherwise, I'm open to suggestions for how to proceed.

Reply:
Are there any other mirroring alternatives available?

Reply:
I've followed up on my initial email to NGA a few times now with no response. Would one possible solution here to label this "latest" and remove the hash? or do this in addition to having a version 3.8 with a hash?

Reply:
@jgsogo 
I don't think we should count on NGA providing/keeping previous releases of geotrans.
We should pro-actively add `geotrans-3.8.tar.gz` to https://github.com/conan-io/cci-sources-backup.
We can add both urls to `conandata.yml` (one to the master.tgz, and one to the tarball in the backup repo);

Reply:
Hi @madebr, unfortunately as per above, the tarball is too large to host on GitHub without having a LFS plan - it's 143MB